src/tools/rados/rados.cc

```diff
@@ -2445,8 +2445,18 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
 
     bufferlist bl;
     if (nargs.size() > min_args) {
-      string val(nargs[min_args]);
-      bl.append(val);
+      if (omap_key.compare("order") == 0 ||
+          omap_key.compare("features") == 0 ||
+          omap_key.compare("size") == 0 ||
+          omap_key.compare("flags") == 0 ||
+          omap_key.compare("snap_seq") == 0 ) {
+        unsigned long long val;
```

Comment:
@tchaikov I just read cls_rbd.cc as your suggestion, and find all these metadata are uint64_t, except "order" is uint8_t, yes, you are right, I will modify it. But it is weird it works ok in my local test...